google/cloud/internal/rest_completion_queue_impl.cc

```diff
@@ -50,11 +50,8 @@ RestCompletionQueueImpl::MakeRelativeTimer(std::chrono::nanoseconds duration) {
 void RestCompletionQueueImpl::RunAsync(
     std::unique_ptr<internal::RunAsyncBase> function) {
   ++run_async_counter_;
-  // Some delay is necessary to reduce the likelihood that the timer expires
-  // before .then() is called.
-  // TODO(#10553): Refactor this mechanism to a more deterministic solution.
-  MakeRelativeTimer(std::chrono::milliseconds(500))
-      .then([f = std::move(function)](auto) { f->exec(); });
+  tq_->Schedule(std::chrono::system_clock::now(),
+                [f = std::move(function)](auto) { f->exec(); });
```

Comment:
Perhaps we should have a `TimerQueue::Schedule(functor)` overload, which would do the same thing but without requiring the caller to utter `now()`.

[Aside: Even if such an overload was just a forwarder, it could avoid `now()` by using a time point like `std::chrono::system_clock::time_point::min()`, or something otherwise known to be in the past.]